File: app/src/main/java/org/digitalcampus/oppia/utils/resources/JSInterfaceForInlineInput.java
package org.digitalcampus.oppia.utils.resources;

import android.content.Context;
import android.util.Log;
import android.webkit.JavascriptInterface;

public class JSInterfaceForInlineInput extends JSInterface{

    private static final String TAG = JSInterfaceForInlineInput.class.getSimpleName();

    //Name of the JS interface to add to the webView
    public static final String INTERFACE_EXPOSED_NAME = "OppiaAndroid_InlineInput";
    private static final String JS_RESOURCE_FILE = "register_inline_input.js";

    private OnInputEnteredListener listener;

    public interface OnInputEnteredListener{
        void inlineInputReceived(String input);
    }

    public JSInterfaceForInlineInput(Context ctx) {
        super(ctx);
        loadJSInjectionSourceFile(JS_RESOURCE_FILE);
    }

    @Override
    public String getInterfaceExposedName() {
        return INTERFACE_EXPOSED_NAME;
    }

    public void setOnInputEnteredListener(OnInputEnteredListener listener){
        this.listener = listener;
    }

    @JavascriptInterface   // must be added for API 17 or higher
    public void registerInlineInput(String userInput) {
        Log.d(TAG, "User input! " + userInput);
        if (listener != null){
            listener.inlineInputReceived(userInput);
        }
    }
}

File: app/src/main/java/org/digitalcampus/oppia/utils/resources/JSInterfaceForInlineInput.kt
package org.digitalcampus.oppia.utils.resources

import android.content.Context
import android.util.Log
import android.webkit.JavascriptInterface
import org.digitalcampus.oppia.listener.OnInputEnteredListener

class JSInterfaceForInlineInput(ctx: Context) : JSInterface(ctx) {

    companion object {
        private val TAG = JSInterfaceForInlineInput::class.simpleName

        //Name of the JS interface to add to the webView
        const val INTERFACE_EXPOSED_NAME = "OppiaAndroid_InlineInput"
        private const val JS_RESOURCE_FILE = "register_inline_input.js"
    }

    private var listener: OnInputEnteredListener? = null

    init {
        loadJSInjectionSourceFile(JS_RESOURCE_FILE)
    }

    override fun getInterfaceExposedName(): String {
        return INTERFACE_EXPOSED_NAME
    }

    fun setOnInputEnteredListener(listener: OnInputEnteredListener) {
        this.listener = listener
    }

    @JavascriptInterface // must be added for API 17 or higher
    fun registerInlineInput(userInput: String) {
        Log.d(TAG, "User input! $userInput")
        listener?.inlineInputReceived(userInput)
    }
}